File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/solutionsvb/vb/workwithsolutions7.vb
' <snippetworkwithsolutions7>


 ' Export or package a solution
 'Export an a solution
 Dim outputDir As String = "C:\temp\"

 Dim exportSolutionRequest As New ExportSolutionRequest()
 exportSolutionRequest.Managed = False
 exportSolutionRequest.SolutionName = solution.UniqueName

 Dim exportSolutionResponse As ExportSolutionResponse =
  CType(_serviceProxy.Execute(exportSolutionRequest), ExportSolutionResponse)

 Dim exportXml() As Byte = exportSolutionResponse.ExportSolutionFile
 Dim filename As String = solution.UniqueName &amp; ".zip"
 File.WriteAllBytes(outputDir &amp; filename, exportXml)

 Console.WriteLine("Solution exported to {0}.", outputDir &amp; filename)

' </snippetworkwithsolutions7>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/solutions/cs/workwithsolutions7.cs
// <snippetworkwithsolutions7>


 // Export or package a solution
 //Export an a solution
 
 ExportSolutionRequest exportSolutionRequest = new ExportSolutionRequest();
 exportSolutionRequest.Managed = false;
 exportSolutionRequest.SolutionName = solution.UniqueName;

 ExportSolutionResponse exportSolutionResponse = (ExportSolutionResponse)_serviceProxy.Execute(exportSolutionRequest);

 byte[] exportXml = exportSolutionResponse.ExportSolutionFile;
 string filename = solution.UniqueName + ".zip";
 File.WriteAllBytes(outputDir + filename, exportXml);

 Console.WriteLine("Solution exported to {0}.", outputDir + filename);

// </snippetworkwithsolutions7>